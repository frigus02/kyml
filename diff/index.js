const fs = require("fs");
const { promisify } = require("util");
const snapshotDiff = require("snapshot-diff");

const readFile = promisify(fs.readFile);

async function getEnvYaml(env) {
  return [
    await readFile(`../examples/${env}/deployment.yml`, "utf8"),
    env !== "feature"
      ? await readFile(`../examples/${env}/ingress.yml`, "utf8")
      : "",
    await readFile(`../examples/${env}/service.yml`, "utf8")
  ].join("\n---\n");
}

test("feature <> staging", async () => {
  const feature = await getEnvYaml("feature");
  const staging = await getEnvYaml("staging");

  expect(
    snapshotDiff(feature, staging, {
      contextLines: 0,
      aAnnotation: "feature",
      bAnnotation: "staging"
    })
  ).toMatchSnapshot();
});

test("staging <> production", async () => {
  const staging = await getEnvYaml("staging");
  const production = await getEnvYaml("production");

  expect(
    snapshotDiff(staging, production, {
      contextLines: 0,
      aAnnotation: "staging",
      bAnnotation: "production"
    })
  ).toMatchSnapshot();
});
