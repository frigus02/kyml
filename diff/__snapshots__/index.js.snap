// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feature <> staging 1`] = `
"Snapshot Diff:
- feature
+ staging

@@ -4,1 +4,1 @@
-   name: the-deployment-BRANCH_NAME
+   name: the-deployment
@@ -7,1 +7,0 @@
-     branch: BRANCH_NAME
@@ -14,1 +13,0 @@
-         branch: BRANCH_NAME
@@ -33,0 +31,16 @@
+ apiVersion: extensions/v1beta1
+ kind: Ingress
+ metadata:
+   name: the-ingress
+   labels:
+     app: hello
+   annotations:
+     nginx.ingress.kubernetes.io/rewrite-target: /
+ spec:
+   rules:
+     - http:
+         paths:
+           - path: /the-path
+             backend:
+               serviceName: the-service
+               servicePort: 80
@@ -38,1 +52,1 @@
-   name: the-service-BRANCH_NAME
+   name: the-service
@@ -41,1 +55,0 @@
-     branch: BRANCH_NAME
"
`;

exports[`staging <> production 1`] = `
"Snapshot Diff:
- staging
+ production

@@ -26,3 +26,1 @@
-               value: \\"User ID=root;Host=the-test-db-server;Port=5432;Database=the-db;\\"
-             - name: DEBUG
-               value: \\"true\\"
+               value: \\"User ID=root;Host=the-db-server;Port=5432;Database=the-db;\\"
"
`;
